### PR TITLE
executor: Provide TIFLASH_INDEXES table

### PR DESCRIPTION
**Pull request by EricZequan**
### What problem does this PR solve?

Issue Number: ref #54245 

Problem Summary:

### What changed and how does it work?

Pick part of https://github.com/tidbcloud/tidb-cse/pull/1030

- supports providing TIFLASH_INDEXES table

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/tables.go [hunk 1]
```diff
@@ -1499,6 +1506,20 @@ var tableTableTiFlashSegmentsCols = []columnInfo{
 	{name: "TIFLASH_INSTANCE", tp: mysql.TypeVarchar, size: 64},
 }
 
+var tableTiFlashIndexesCols = []columnInfo{
+	{name: "TIDB_DATABASE", tp: mysql.TypeVarchar, size: 64},
+	{name: "TIDB_TABLE", tp: mysql.TypeVarchar, size: 64},
+	{name: "TABLE_ID", tp: mysql.TypeLonglong, size: 64},
```
Comment (zimulala) [inline pkg/infoschema/tables.go:1512]: 64->21 or update when I handle https://github.com/pingcap/tidb/issues/48323？

Comment (EricZequan) [reply pkg/infoschema/tables.go]: modified.

**Review by zimulala — APPROVED** [review]
LGTM

**Comment by breezewish** [comment]
Partially duplicated with https://github.com/pingcap/tidb/pull/55817?

**Comment by EricZequan** [comment]
> Partially duplicated with #55817?

image

#55817 not pick `tiflash_indexex` table

**Review by breezewish — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**